woocommerce/wc-admin
# Add paypal commerce obw integration

Fixes #5788

This PR replaces the use of the Paypal Express Checkout plugin with the Paypal payments plugin.
It uses the new connect method with the plugin that shows a new popup setting a user up with Paypal.
There is a fallback incase something fails to add the necessary fields manually.

### Accessibility


### Screenshots

https://d.pr/v/t5KHfP

Demo of the fallback more manual approach -> https://d.pr/v/Oq2HAp

### Detailed test instructions:

**If the Paypal Payments plugin isn't updated yet** pull down the [Paypal Payments plugin repo](https://github.com/woocommerce/woocommerce-paypal-payments) into your plugins folder and [setup the local environment](https://github.com/woocommerce/woocommerce-paypal-payments#development).
You can test it first with `trunk`, which should show the manual input fields (if `obw-work` isn't merged yet), then test it with the branch [obw-work](https://github.com/woocommerce/woocommerce-paypal-payments/compare/obw-work) which should show the Paypal modal window and connect button.
When switching branches you have to run `composer dumpautoload` and also `npm run build:dev`.
You can create a zip by running `npm run build` if testing on a hosted platform.

- Create new Woo store and finish the onboarding wizard
- Go to the **Set up payments** task and click **Set up** on the Paypal plugin. (It should say Paypal Payments)
- It should automatically start the **Install** step, and install the [Paypal Payments](https://woocommerce.com/products/woocommerce-paypal-payments/) plugin.
- Once step two is loaded it should show a `Connect` button (relies on the Paypal plugin being updated)
  - If paypal plugin is the old one, it will show the manual fields (email, merchant id, client id, client secret). 
- Click on **Connect** a window should popup for Paypal, follow this until finished. The last button is - Go back to Woocommerce Developers
- Once done, the page should reload, and briefly show the setup screen again, it should then finish and go back to the payment list.
- Once on the payment list, the `Set up` button should be gone, and instead show a toggle, that is set to enabled.
  - The enable/disable button should be correctly reflected in the Woocommerce payment settings screen as well.

Quick client test (With paypal payments enabled)
- Make sure you have added a product and store homescreen
- Navigate to one of the products and add it to the cart
- Click **go to checkout**
- Paypal should be one of the options to pay
- Filling in your billing/shipping info then click pay with **Paypal**
- The paypal pay window should pop up correctly without errors.
